dynamo/read.js

```diff
@@ -54,45 +51,51 @@ const rangeScan = function (email, start, end) {
       ":end": end,
     },
   };
-  return new Promise((resolve, reject) => {
-    docClient.query(params, function (err, data) {
-      if (err) return reject("gmail-stats::rangeScan::error - " + err);
-      if (
-        !data.Items ||
-        data.Items.length === 0 ||
-        !data.Items[0].hasOwnProperty("id")
-      )
-        return reject("gmail-stats::rangeScan::keyNotFound");
-      // Step 0: Initialize variables
-      const email = data.Items[0].id;
-      const times = [];
-      const toMeFromGmail = [];
-      const toMeFromNonGmail = [];
-      const fromMeToGmail = [];
-      const fromMeToNonGmail = [];
+  try {
+    const data = await docClient.query(params).promise();
+    if (
+      !data.Items ||
+      data.Items.length === 0 ||
+      !data.Items[0].hasOwnProperty("id")
+    )
+      throw "gmail-stats::rangeScan::keyNotFound";
 
-      // Step 1: Iterate through Items, filtering, sorting, then extracting
-      let items = data.Items;
-      for (gmailItem of items) {
-        times.push(gmailItem.time);
-        toMeFromGmail.push(gmailItem.toMeFromGmail);
-        toMeFromNonGmail.push(gmailItem.toMeFromNonGmail);
-        fromMeToGmail.push(gmailItem.fromMeToGmail);
-        fromMeToNonGmail.push(gmailItem.fromMeToNonGmail);
-      }
+    // Step 0: Initialize variables
+    const email = data.Items[0].id;
+    const times = [];
+    const toMeFromGmail = [];
+    const toMeFromNonGmail = [];
+    const fromMeToGmail = [];
+    const fromMeToNonGmail = [];
 
-      // Step 2: Resolve!
-      resolve({
-        email,
-        times,
-        toMeFromGmail,
-        toMeFromNonGmail,
-        fromMeToGmail,
-        fromMeToNonGmail,
-      });
-    });
-  });
+    // Step 1: Extract
+    let items = data.Items;
+    for (gmailItem of items) {
+      times.push(gmailItem.time);
+      toMeFromGmail.push(gmailItem.toMeFromGmail);
+      toMeFromNonGmail.push(gmailItem.toMeFromNonGmail);
+      fromMeToGmail.push(gmailItem.fromMeToGmail);
+      fromMeToNonGmail.push(gmailItem.fromMeToNonGmail);
+    }
+
+    // Step 2: Resolve!
+    return {
+      email,
+      times,
+      toMeFromGmail,
+      toMeFromNonGmail,
+      fromMeToGmail,
+      fromMeToNonGmail,
+    };
+  } catch (err) {
+    throw "gmail-stats::rangeScan::error - " + err;
+  }
 };
 
 exports.read = fetchOneByKey;
 exports.rangeScan = rangeScan;
+
+exports
+  .rangeScan("micahtyong@gmail.com", 1604707200, 1606676400)
```

Comment:
I'm guessing this is test code?

Reply:
Yup, I'll get rid of that. I'll probably add a linter soon as I'm forgetting to remove the scattered test calls and console.log commands throughout the codebase.